### PR TITLE
Added SPDX compliance validation and tests

### PR DESCRIPTION
**Pull request by disulliv**
Solution to issue #756

### CODE REVIEW

File: test/fixtures/curation-valid.1.yaml [hunk 1]
```diff
@@ -2,40 +2,40 @@ coordinates:
...
-          - 'foo'
-          - 'bar'
+          - "foo"
+          - "bar"
         doc:
-          - 'foo'
+          - "foo"
     licensed:
-      declared: 'MIT'
+      declared: "MIT"
```
Comment (jeffmendoza) [inline test/fixtures/curation-valid.1.yaml:41]: It would be good to test something complex for the positive case, maybe multiple and/or with parenthesis.

Comment (disulliv) [reply test/fixtures/curation-valid.1.yaml:41]: Sure, I can add a case for that

Comment (disulliv) [reply test/fixtures/curation-valid.1.yaml:41]: Added `license: "MIT AND (LGPL-2.1-or-later OR BSD-3-Clause)"`

**Review by jeffmendoza — APPROVED** [review]

**Review by nellshamrell — APPROVED** [review]
Well done! Feel free to merge and deploy!

**Pull request merged**